docs/design/StratisSoftwareDesign.lyx

```diff
@@ -6933,5 +6933,423 @@ Based on looking at the existing available building blocks, the best option
+ an internal name.
+\end_layout
+
+\begin_layout Standard
+\begin_inset listings
+inline false
+status collapsed
+
+\begin_layout Plain Layout
+
+<KEY_DESCRIPTION>
+\end_layout
+
+\end_inset
+
+
+\end_layout
+
+\begin_layout Standard
+The key description in this token is not used for unlocking, only to refer
+ back to the corresponding LUKS2 keyring token which handles unlocking the
+ device.
+ The key description in both should be identical.
+\end_layout
+
+\begin_layout Standard
+\begin_inset listings
+inline false
+status collapsed
+
+\begin_layout Plain Layout
+
+<POOL_UUID>
+\end_layout
+
+\end_inset
+
+
+\end_layout
+
+\begin_layout Standard
+This UUID corresponds to the Stratis pool that owns this block device.
+ The pool UUID in the token matches the pool UUID in the encrypted Stratis
+ metadata.
+\end_layout
+
+\begin_layout Standard
+\begin_inset listings
+inline false
+status collapsed
+
+\begin_layout Plain Layout
+
+<DEVICE_UUID>
+\end_layout
+
+\end_inset
+
+
+\end_layout
+
+\begin_layout Standard
+This UUID corresponds to the device UUID for this specific block device.
+ The device UUID in the token matches the device UUID in the encrypted Stratis
+ metadata.
+\end_layout
+
+\begin_layout Subsubsection
+Encrypted Stratis metadata format
+\end_layout
+
+\begin_layout Standard
+Once the LUKS2 volume is unlocked, the encrypted Stratis metadata should
+ be accessible through the unencrypted logical devicemapper device exactly
+ the same way as an unencrypted Stratis device.
+\end_layout
+
+\begin_layout Subsection
+Encrypted device creation
+\end_layout
+
+\begin_layout Standard
+Stratis does not wipe the device prior to initializing it with a LUKS2 header.
+ Due to how data is encrypted with LUKS2, if a user is concerned with an
+ attacker being able to determine exactly how much of the data on the device
+ is encrypted, the device should be wiped with random data prior to providing
+ it to a Stratis pool.
+ If a user does not want data that was previously stored on the device (encrypte
+d or unencrypted) to be visible, the disk should be zeroed or wiped with
+ random data.
+\end_layout
+
+\begin_layout Subsection
+Encrypted device discovery
+\end_layout
+
+\begin_layout Standard
+To find encrypted Stratis devices, stratisd searches for LUKS2 volumes present
+ on the system and validates that the LUKS2 metadata on the device is well
+ formed LUKS2 metadata laid down by Stratis.
+ Once this has been verified, Stratis stores a record of this device and
+ waits for the user to activate the LUKS2 volume which can be initiated
+ by a D-Bus API method.
+ The devices will be unlocked with the corresponding passphrase in the kernel
+ keyring as specified by the key description in the LUKS2 keyring token.
+ Stratis will then search all resulting unencrypted logical devices for
+ the Stratis metadata and will construct pools accordingly.
+\end_layout
+
+\begin_layout Subsection
+Encrypted device activation
```

Comment:
Please also add in this section a remark about the fact that, as a consequence of this design, a Stratis device can be opened not-by-Stratis so long as the key is in the keyring.